src/ThreeEditor/components/Dialog/SaveFileDialog.tsx

```diff
@@ -1,35 +1,37 @@
-import {
-	Button,
-	Checkbox,
-	Dialog,
-	DialogActions,
-	DialogContent,
-	DialogContentText,
-	FormControlLabel,
-	TextField
-} from '@mui/material';
+import { Button, Checkbox, FormControlLabel, TextField } from '@mui/material';
 import { ChangeEvent, useEffect, useState } from 'react';
 
 import { FullSimulationData } from '../../../services/ShSimulatorService';
+import { saveString } from '../../../util/File';
 import { YaptideEditor } from '../../js/YaptideEditor';
-import { CustomDialogTitle } from './CustomDialog';
+import { CustomDialog, WarnDialogProps } from './CustomDialog';
 
-export type SaveFileProps = {
-	open: boolean;
-	onClose: () => void;
-	onConfirm: (data: {}, fileName: string) => void;
-	editor: YaptideEditor;
+const saveJson = (data: {}, fileName: string) => {
+	let output = undefined;
+	try {
+		output = JSON.stringify(data, null, '\t');
+		output = output.replace(/[\n\t]+([\d.e\-[\]]+)/g, '$1');
```

Comment:
The regular expression is defined as /[\n\t]+([\d.e\-[\]]+)/g. The first part of the regular expression, [\n\t]+, matches one or more occurrences of either a newline or a tab character. The second part of the regular expression, ([\d.e\-[\]]+), matches one or more occurrences of a numeric value, including scientific notation and negative values. The parentheses around this part of the regular expression capture the matched value so that it can be used in the replacement string.

The regular expression is used with the replace method to replace any occurrence of one or more newlines or tabs followed by a numeric value with just the numeric value. The replacement string is $1, which refers to the captured numeric value from the regular expression.